src/base/bittorrent/dbresumedatastorage.cpp

```diff
@@ -507,8 +516,10 @@ void BitTorrent::DBResumeDataStorage::createDB() const
     }
 }
 
-void BitTorrent::DBResumeDataStorage::updateDBFromVersion1() const
+void BitTorrent::DBResumeDataStorage::updateDB(int fromVersion) const
```

Comment:
```suggestion
void BitTorrent::DBResumeDataStorage::updateDB(const int fromVersion) const
```

Reply:
It is changed in function body after intermediate updates.

Reply:
Why not make some changes to the code using it and make it immutable?

Reply:
> Why not make some changes to the code using it and make it immutable?

Currently the update process is handled sequentially. E.g. if I need to update from v1 to v3 I do it step by step:
1. v1 -> v2
2. v2 -> v3

If I need to update from v2 to v3 I just skip the step 1 above.
To handle this naturally I just update `fromVersion` after each step.
I don't think I can keep the algorithm simple/clear by making `fromVersion` constant. Of course I can add yet another mutable variable but it has no sense since there is one.

Reply:
Well, I've found a way to satisfy you.

Reply:
>>Why not make some changes to the code using it and make it immutable?
>
>Currently the update process is handled sequentially. 

I recognized this pattern, it is similar to what is done in upgrade.cpp and therefore I reckon it should be possible.